### PR TITLE
Expand GitHub service methods

### PR DESCRIPTION
**Pull request by courte**
Adds functionality to @nahurst's awesome contribution in #268 to allow for:
- grabbing issues and forks for repos
- grabbing pull requests, commits, issues, and forks for users 

The repetition in the code pains me. There must be a less repetitive way to create these queries and refactor this. But alas, this way seems to work.

### CODE REVIEW

**Comment by courte** [comment]
@DBNess: confirmation that #300 is working!

**Comment by courte** [comment]
TravisCI is not passing. API get request uri is different sometimes (in this case, it is missing the "&per_page=100" section at the end). Easy way out is to set vcr to `record: :new_episodes`, buuuuut also why is it doing that? Why?

File: app/services/github.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    results = Octokit.search_repos "user:#{user} fork:only created:#{day_begin}..#{day_end}"
+    results.items
+  end
+
+  def org_repo(org, repo)
+    "#{org}/#{repo}"
+  end
+
+  def parse_org_repo(url)
+    pr_url_regex = /\Ahttps:\/\/api.github.com\/repos\/([\w\-]+)\/([\w\-]+)\/issues\/\d+\z/
```
Comment (DBNess) [inline app/services/github.rb]: Do we have this exact code somewhere else, too?

File: app/services/github.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  def forks_by_user(user, day_begin, day_end)
+    results = Octokit.search_repos "user:#{user} fork:only \
+                                    created:#{day_begin}..#{day_end}"
+    results.items
+  end
+
+  def org_repo(org, repo)
+    "#{org}/#{repo}"
+  end
```
Comment (DBNess) [inline app/services/github.rb]: this `org_repo` method doesn't seem like it actually saves us any keystrokes or adds any clarity -- do we need it?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -16,11 +17,17 @@
...
 require 'rspec/autorun'
 require 'capybara/rspec'
+require 'vcr'
 
 # Requires supporting ruby files with custom matchers and macros, etc,
 # in spec/support/ and its subdirectories.
 Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
 
+VCR.config do |c|
+  c.cassette_library_dir     = 'spec/cassettes'
```
Comment (DBNess) [inline spec/spec_helper.rb]: WOohoooo :100: for #300 amirite? Nice job, @courte!

**Comment by DBNess** [comment]
:+1: this is a ton of cassettes and we expect to refactor the major service methods, but I like where this is going! :truck: movin along

**Comment by DBNess** [comment]
Just wanted to formally :+1: :+1: :+1: :sunglasses: though this is covered by #306

**Pull request merged**